### ISSUE TITLE
conda doesn't use anaconda-client url

### ISSUE THREAD

**Issue by postelrich** [conda/conda#3442]
I started to get errors with my CI when it came to the conda build step.  After digging, with @msarahan help, discovered that the step that install conda-build was also updating conda to 4.2.6.  This version appears to have issues using the url set via anaconda-repo.  Was able to reproduce locally too.

```
anaconda config --set url <private anaconda repo>/api
anaconda login
conda install channelpackage -c mychannel
```

throws the following error:

```
○ → conda install channelpackage -c mychannel
WARNING conda.connection:get_binstar_token(106): Warning: could not capture token from anaconda-client (RecursionError('maximum recursion depth exceeded while calling a Python object',))
WARNING conda.connection:get_binstar_token(106): Warning: could not capture token from anaconda-client (RecursionError('maximum recursion depth exceeded while calling a Python object',))
CondaHTTPError: HTTP Error: Could not find URL: https://conda.anaconda.org/mychannel/linux-64/
```

**Comment by kalefranz** [comment]
Can you give the output of 

    conda install channelpackage -c mychannel -vv

**Comment by kalefranz** [comment]
Thanks for the report btw.

**Comment by kalefranz** [comment]
Also, might have to get help from @dsludwig here.  The debug output with `-vv` will help though...

**Comment by postelrich** [comment]
Here's the tail end of it:

```
Invalid Token

DEBUG conda.fetch:fetch_repodata(82): Opening repodata cache for https://conda.anaconda.org/mychannel/noarch/ at /home/ec2-user/miniconda/pkgs/cache/88ae923c.json
DEBUG conda.connection:get_binstar_token(86): Attempting to binstar collect token for url https://conda.anaconda.org/mychannel/noarch/repodata.json
DEBUG conda.connection:add_binstar_token(78): Adding binstar token to url https://conda.anaconda.org/mychannel/noarch/repodata.json
DEBUG requests.packages.urllib3.connectionpool:_make_request(395): "GET /t/su-fde61afe-06b5-42ae-aea5-4bf4b92ab064/mychannel/noarch/repodata.json HTTP/1.1" 401 13
DEBUG conda.fetch:fetch_repodata(114):
> GET /t/su-fde61afe-06b5-42ae-aea5-4bf4b92ab064/mychannel/noarch/repodata.json HTTPS
> User-Agent: conda/4.2.6 requests/2.10.0 CPython/3.5.2 Linux/3.10.0-327.22.2.el7.x86_64 Red Hat Enterprise Linux Server/7.2 glibc/b'2.17'
> Accept: */*
> Accept-Encoding: gzip, deflate, compress, identity
> Connection: keep-alive
> Content-Type: application/json
> Cookie: session=eyJfaWQiOnsiIGIiOiJORE15TkRabVlUTmlOMlppTnpFek0yWmhPR1U1TnpBME56azNNek0xTXpRPSJ9fQ.CryN2A.b7X0JciJZTNu6TPfC2MY6HmNeUQ

< HTTPS 401 UNAUTHORIZED
> Content-Type: text/html; charset=utf-8
> Date: Thu, 15 Sep 2016 19:54:00 GMT
> license_expired: False
> license_valid: True
> Server: nginx/1.8.1
> Set-Cookie: session=eyJfaWQiOnsiIGIiOiJORE15TkRabVlUTmlOMlppTnpFek0yWmhPR1U1TnpBME56azNNek0xTXpRPSJ9fQ.CryN2A.b7X0JciJZTNu6TPfC2MY6HmNeUQ; Domain=.anaconda.org; HttpOnly; Path=/
> Content-Length: 13
> Connection: keep-alive

Invalid Token

DEBUG conda.fetch:fetch_repodata(82): Opening repodata cache for https://conda.anaconda.org/mychannel/noarch/ at /home/ec2-user/miniconda/pkgs/cache/88ae923c.json
DEBUG conda.connection:get_binstar_token(86): Attempting to binstar collect token for url https://conda.anaconda.org/mychannel/noarch/repodata.json
WARNING conda.connection:get_binstar_token(106): Warning: could not capture token from anaconda-client (RecursionError('maximum recursion depth exceeded while calling a Python object',))
DEBUG requests.packages.urllib3.connectionpool:_make_request(395): "GET /mychannel/noarch/repodata.json HTTP/1.1" 404 2298
DEBUG conda.fetch:fetch_repodata(114):
> GET /mychannel/noarch/repodata.json HTTPS
> User-Agent: conda/4.2.6 requests/2.10.0 CPython/3.5.2 Linux/3.10.0-327.22.2.el7.x86_64 Red Hat Enterprise Linux Server/7.2 glibc/b'2.17'
> Accept: */*
> Accept-Encoding: gzip, deflate, compress, identity
> Connection: keep-alive
> Content-Type: application/json
> Cookie: session=eyJfaWQiOnsiIGIiOiJORE15TkRabVlUTmlOMlppTnpFek0yWmhPR1U1TnpBME56azNNek0xTXpRPSJ9fQ.CryN2A.b7X0JciJZTNu6TPfC2MY6HmNeUQ

< HTTPS 404 NOT FOUND
> Content-Encoding: gzip
> Content-Type: text/html; charset=utf-8
> Date: Thu, 15 Sep 2016 19:54:00 GMT
> license_expired: False
> license_valid: True
> Server: nginx/1.8.1
> Set-Cookie: session=.eJw9zM0KgkAUhuFbibNuoRNuBBfBmCicbxAUcTZCP9I0WFBBecR7b1Zt3xeehQZ3pnShzZFSguYZ2k59AwdlHSQXVvZqijaB7COI37FwzFJntG7p9HqOw_vhL_c_Ybs6hi4jO7VfyMFzUXno6oap_JiO59CVacJTvbINPIo8QZ0Fbv0BKBIshQ.CryN2A.QjoneAgwnyyB0jKJyUWYx1vO9rs; Domain=.anaconda.org; HttpOnly; Path=/
> Content-Length: 2298
> Connection: keep-alive

<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1, maximum-scale=1">

    <title>Not Found :: Anaconda Cloud</title>
    <link rel="shortcut icon" href="//binstar-static-prod.s3.amazonaws.com/latest/img/apple-touch-icon.png">
    <link rel="stylesheet" type="text/css" href="//binstar-static-prod.s3.amazonaws.com/latest/css/app.css">
    <!--[if gte IE 9]>
      <style type="text/css">
        .gradient {
           filter: none;
        }
      </style>
    <![endif]-->
    <script src="//binstar-static-prod.s3.amazonaws.com/latest/js/bundle.js"></script>


    <!-- Google Tag Manager -->
<noscript><iframe src="//www.googletagmanager.com/ns.html?id=GTM-MTFQWW"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-MTFQWW');</script>
<!-- End Google Tag Manager -->




  </head>
  <body >

      <div class="contain-to-grid">
  <nav class="top-bar" data-topbar role="navigation">
    <ul class="title-area">
      <li class="name">
        <h1>
          <a class="navbar-brand" href="/">
            <img src="//binstar-static-prod.s3.amazonaws.com/latest/img/logo-white.svg"/>
          </a>
        </h1>
      </li>
      <li class="toggle-topbar menu-icon"><a href="#"><span>Menu</span></a></li>
    </ul>
    <section class="top-bar-section">
      <ul class="right">
        <li class="has-form show-for-small-only">
          <form action="https://anaconda.org/search">
            <div class="row collapse">
              <div class="small-9 columns">
                <input type="text" placeholder="Search Anaconda Cloud" name="q">
              </div>
              <div class="small-3 columns">
                <button type="submit" class="postfix"><i class="fa fa-search"></i></button>
              </div>
            </div>
          </form>
        </li>

        <li><a href="https://anaconda.org/gallery">Gallery</a></li>


        <li><a href="https://anaconda.org/about">About</a></li>

        <li><a href="https://anaconda.org/about/pricing">Pricing</a></li>

        <li><a href="http://continuum.io/anaconda" target="_blank">Anaconda</a></li>
        <li><a href="https://docs.continuum.io/anaconda-cloud/" target="_blank">Help</a></li>

        <li><a href="http://continuum.io/downloads" target="_blank">Download Anaconda</a></li>

        <li><a href="https://anaconda.org/account/login?next=%2Fmychannel%2Fnoarch%2Frepodata.json">Sign In</a></li>
      </ul>
    </section>
  </nav>
</div>

    <div class="container">

      <div class="row">
        <div class="columns medium-12">




        </div>
      </div>

        <div class="row"><div class="columns medium-12">
<div class="row">
  <div class="columns medium-6 medium-offset-3 end text-center">
    <div class="panel callout">
      <h3>

      </h3>
      <p>

<h1>That&#8217s a 404, folks</h1>
<h3>I think you went to the wrong page.</h3>

<p>

  <form action="https://anaconda.org/account/login?next=https%3A%2F%2Fconda.anaconda.org%2Fmychannel%2Fnoarch%2Frepodata.json" method="post">
    <div style="display:none;"><input id="csrf_token" name="csrf_token" type="hidden" value="1473972840##0537429414e1bbc995f61f5afe21c1476869e00b"></div>
    <input id="username" name="username" placeholder="Username" class="" type="text" value="" >
  <small id="username-error"class="error" style="display: none"></small>

  <input id="password" name="password" placeholder="Password" class="" type="password" value="" >
  <small id="password-error"class="error" style="display: none"></small>

    <button class="expand" type="submit">Log In</button>
  </form>

</p>
<p>

  <a href='/account/forgot_password'>I forgot my password.</a>


  <br>
  <a href="https://anaconda.org/account/register">Register for an account</a>

</p>



      </p>
    </div>
  </div>
</div>
</div></div>

      <div class="push"></div>
    </div>
    <footer class="">

  <div class="row">
    <div class="columns medium-3 medium-offset-3">
      <h5>Anaconda Cloud</h5>
      <p>
        <a href="https://anaconda.org/gallery">Gallery</a><br>
        <a href="https://anaconda.org/about">About</a><br>

        <a href="https://anaconda.org/about/pricing">Pricing</a><br>

        <a href="https://docs.continuum.io/anaconda-cloud/" target="_blank">Documentation</a><br>
        <a href="https://anaconda.org/contact">Support</a><br>
        <a href="https://www.continuum.io/our-story" target="_blank">About Continuum</a><br>
        <a href="http://continuum.io/downloads" target="_blank">Download Anaconda</a><br>
        <span>Anaconda Cloud v2.20.2</span>
      </p>
    </div>
    <div class="columns medium-3 end">
      <h5>Community</h5>
      <p>
        <a href="https://www.continuum.io/content/anaconda-community" target="_blank">Anaconda Community</a><br>
        <a href="https://www.continuum.io/open-source-core-modern-software" target="_blank">Open Source</a><br>
        <a href="https://www.continuum.io/content/numfocus" target="_blank">NumFOCUS</a><br>
        <a href="https://www.continuum.io/content/why-python" target="_blank">Why Python</a><br>
        <a href="https://www.continuum.io/support" target="_blank">Support</a><br>
        <a href="https://www.continuum.io/blog/developer-blog" target="_blank">Developer Blog</a><br>
        <a href="https://www.continuum.io/snapshot-our-contributors" target="_blank">Our Contributors</a>
      </p>
    </div>
  </div>
  <br>
  <div class="row">
    <div class="columns medium-12">
      <ul>
        <li><a href="https://twitter.com/ContinuumIO" target="_blank"><i class="fa fa-2x fa-twitter"></i></a></li>
        <li><a href="https://www.facebook.com/ContinuumAnalytics" target="_blank"><i class="fa fa-2x fa-facebook"></i></a></li>
        <li><a href="http://www.slideshare.net/continuumio" target="_blank"><i class="fa fa-2x fa-slideshare"></i></a></li>
        <li><a href="https://www.linkedin.com/company/continuum-analytics-inc-" target="_blank"><i class="fa fa-2x fa-linkedin"></i></a></li>
        <li><a href="https://instagram.com/continuumanalytics/" target="_blank"><i class="fa fa-2x fa-instagram"></i></a></li>
        <li><a href="https://www.youtube.com/c/ContinuumIo" target="_blank"><i class="fa fa-2x fa-youtube"></i></a></li>
        <li><a href="https://github.com/ContinuumIO" target="_blank"><i class="fa fa-2x fa-github"></i></a></li>
      </ul>
    </div>
  </div>

</footer>
<div class="license full-width">
  <div class="row">
    <div class="columns medium-12 text-center">
      <h5>

        <a href="https://anaconda.org/about/legal/privacy" target="_blank">PRIVACY POLICY</a>
        &nbsp;&#124;&nbsp;
        <a href="https://anaconda.org/about/legal/terms" target="_blank">EULA</a>

        &nbsp; &nbsp; &nbsp; &copy;
        2016 Continuum Analytics, Inc. All Rights Reserved.
      </h5>
    </div>
  </div>
</div>


  </body>
</html>

CondaHTTPError: HTTP Error: Could not find URL: https://conda.anaconda.org/mychannel/linux-64/
```

**Comment by kalefranz** [comment]
Can you try

    conda config --set channel_alias <private anaconda repo>/api

or

    conda config --set channel_alias <private anaconda repo>

?

Curious if either of those fixes the problem.  Not that it's an optimal solution.  We can do better.

**Comment by kalefranz** [comment]
addressed in #3550

**Comment by kalefranz** [comment]
@postelrich 

Conda 4.2.10 is available on the `conda-canary` channel, and it should have a fix for this issue.  If you have time, please help test it.

    conda config --prepend channels conda-canary
    conda update conda

We'll likely be releasing this build to the general availability defaults channel on Friday.
